### PR TITLE
[WL refactor] simplify (dis)connection logic & deprecate redundant types

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

- deprecate `WalletLinkResponseEventData`
- cleanup `WalletLinkSession`
- derive `Web3Method` from `Web3Request`, and remove the legacy one
- remove scanQRCode type
- encapsulate & simplify `connection.destroy()`
- deprecate `WalletLinkSessionConfig`
- deprecate `connection.setMetadata(...)`
- deprecate `onceConnected`
- simplify post-connect logic

### _How did you test your changes?_

unit tests

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/sign/walletlink/relay/connection/WalletLinkConnection.ts [hunk 1]
```diff
@@ -173,16 +174,24 @@ export class WalletLinkConnection {
...
+
+    await this.makeRequest(
+      {
+        type: 'SetSessionConfig',
+        id: IntNumber(this.nextReqId++),
+        sessionId: this.session.id,
+        metadata: { __destroyed: '1' },
+      },
+      { timeout: 1000 }
+    );
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/relay/connection/WalletLinkConnection.ts:188]: image

Comment (bangtoven) [reply packages/wallet-sdk/src/sign/walletlink/relay/connection/WalletLinkConnection.ts:188]: image

File: packages/wallet-sdk/src/sign/walletlink/relay/type/Web3Method.ts [hunk 2]
```diff
@@ -1,18 +0,0 @@
-// Copyright (c) 2018-2023 Coinbase, Inc. <https://www.coinbase.com/>
-
-export const web3Methods = [
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/relay/type/Web3Method.ts:3]: derive Web3Method from Web3Request, and remove the legacy one

File: packages/wallet-sdk/src/sign/walletlink/relay/WalletLinkRelay.ts [hunk 3]
```diff
@@ -137,27 +135,19 @@ export class WalletLinkRelay implements WalletLinkConnectionUpdateListener {
...
     }
   };
 
   public resetAndReload(): void {
-    Promise.race([
-      this.connection.setSessionMetadata('__destroyed', '1'),
-      new Promise((resolve) => setTimeout(() => resolve(null), 1000)),
-    ])
+    this.connection
+      .destroy()
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/relay/WalletLinkRelay.ts:149]: encapsulate all these complexity into `.destroy()`

**Review by fan-zhang-sv — APPROVED** [review]

**Pull request merged**